src/com/facebook/buck/apple/project_generator/SchemeGenerator.java

```diff
@@ -301,6 +327,13 @@ public static Element serializeTestAction(Document doc, XCScheme.TestAction test
       testableElem.appendChild(refElem);
     }
 
+    if (testAction.getEnvironmentVariables().isPresent()) {
+      testActionElem.setAttribute("shouldUseLaunchSchemeArgsEnv", "NO");
```

Comment:
Is there any reason why you disable it here?
Maybe you could add a comment?

Reply:
On line 316, this is set to yes by default. So by default the test action will have the Launch actions environment variables. If however, you specify custom variables for the test action, you should not be using the launch action's variables here.

In xcode, shouldUseLaunchSchemeArgsEnv controls a checkbox that greys out the custom environment variables. This is an override that is enabled by default, so to have the custom variables we need to make sure the checkbox is not enabled.